### PR TITLE
Implement named pipes

### LINKED ISSUE #4: Feature Request: Named pipes
Could we use posix FIFOs to get the console output of csgo and pump it into csgo-csmd, without having to call it, making it so one doesn't even need to tab out of the game?

### PR DESCRIPTION
**Pull request by Cyclic3**
(Fixes #4)

I will be away for the next week, so I will be unable to work on this. Here is my work so far.

It works fine on my machine (TM), but I have only tested it on 3 different servers.

I have added a flag in the config file that can be used to enable the fifo code (`use_fifo`), which is off by default.

### CODE REVIEW

**Comment by M-Reimer** [comment]
I'm still unsure if this is something worth to make code even complexer.

- It is still needed to call the script which I won't do every time I play CS as maybe I plan to just play some deathmatch.
- If I later decide that I want to play community servers, then I may have to tab out anyway to start the script.
- Even if I play community servers, at some point I'll have all files for my favorite servers on disc and don't need the script running at all.
- If the script only runs in background and I want to join a server which needs a bigger list of files, then at some point I'll tab out to see the download progress.
- It sucks to just wait for something without any progress indication at all.

IMHO it would be a much nicer idea to clean up the script output a bit (get proper progress bars).

**Comment by Cyclic3** [comment]
@M-Reimer I suppose this is where we differ:
* I launch csgo from the command line meaning that I can integrate a script much easier
* Therefore launching is a non-issue for me
* For some unknown reason, cs only works properly on a ram disk for me (https://github.com/ValveSoftware/csgo-osx-linux/issues/1654), and so I need to redownload maps every reboot. Also many of the servers I go on have large map pools, meaning that there is a good chance I have not seen the map before. These combined mean that, for me, having all files downloaded is not going to happen.
* Agreed. But I have a multi-monitor setup, which means I do not need to tab out to look at something.

In conclusion, I will maintain my own fork with these features, since whilst I need them, most others will not.

If you want to close this, feel free.
